share/random_namespace.go

```diff
@@ -38,22 +39,55 @@ func RandomBlobNamespace() Namespace {
 	for {
 		id := RandomBlobNamespaceID()
 		namespace := MustNewV0Namespace(id)
-		if isBlobNamespace(namespace) {
+		if IsBlobNamespace(namespace) {
 			return namespace
 		}
 	}
 }
 
-// isBlobNamespace returns an true if this namespace is a valid user-specifiable
-// blob namespace.
-func isBlobNamespace(ns Namespace) bool {
-	if ns.IsReserved() {
-		return false
+// AddInt adds arbitrary int value to namespace, treating namespace as big-endian
+// implementation of int
+func AddInt(n Namespace, val int) (Namespace, error) {
```

Comment:
[question] why does celestia-node need to add ints to a namespace? It seems confusing that this library exposes a method to add an integer to a type (namespace) that doesn't really support that.

If you need a way to generate namespaces that are each one after the other, I propose we create a different utility method that generates and returns ascending namespaces instead of adding this function.

Reply:
we need it for some tricky test cases in the node. I propose to explicitly add `t* testing.T` to the argument list

Reply:
Yea that works. Also could help to add a Godoc like this:

```go
// AddInt adds arbitrary int value to namespace, treating namespace as big-endian
// implementation of int. Note: should only be used for tests.
func AddInt(t *testing.T, n Namespace, val int) (Namespace, error) {
```